NitroxClient/MonoBehaviours/Vehicles/CyclopsMovementReplicator.cs

```diff
@@ -0,0 +1,88 @@
+using NitroxClient.GameLogic;
+using NitroxModel.Packets;
+using UnityEngine;
+
+namespace NitroxClient.MonoBehaviours.Vehicles;
+
+public class CyclopsMovementReplicator : VehicleMovementReplicator
+{
+    private SubControl subControl;
+
+    private RemotePlayer drivingPlayer;
+    private bool throttleApplied;
+    private float steeringWheelYaw;
+
+    public void Awake()
+    {
+        subControl = GetComponent<SubControl>();
+    }
+
+    public new void Update()
+    {
+        base.Update();
+
+        if (subControl.canAccel && throttleApplied)
+        {
+            // See SubControl.Update
+            var topClamp = subControl.useThrottleIndex switch
+            {
+                1 => 0.66f,
+                2 => 1f,
+                _ => 0.33f,
+            };
+            subControl.engineRPMManager.AccelerateInput(topClamp);
+            for (int i = 0; i < subControl.throttleHandlers.Length; i++)
+            {
+                subControl.throttleHandlers[i].OnSubAppliedThrottle();
+            }
+        }
+
+        if (Mathf.Abs(steeringWheelYaw) > 0.1f)
+        {
+            ShipSide shipSide = steeringWheelYaw > 0 ? ShipSide.Port : ShipSide.Starboard;
+            for (int i = 0; i < subControl.turnHandlers.Length; i++)
+            {
+                subControl.turnHandlers[i].OnSubTurn(shipSide);
+            }
+        }
+    }
+
+    public override void ApplyNewMovementData(MovementData newMovementData)
+    {
+        if (newMovementData is not DrivenVehicleMovementData vehicleMovementData)
+        {
+            return;
```

Comment:
This is no issue actually. This step only happens after the interpolation has happened and is only there in case the vehicle is being driven. But we also receive the movement update packets when not driving a vehicle. For example when you stop driving a cyclops but it still has some thrust that drives in you some direction, this movement is still broadcasted to all players.